lib/pure/random.nim

```diff
@@ -635,20 +653,60 @@ when not defined(nimscript) and not defined(standalone):
       let time = int64(times.epochTime() * 1000) and 0x7fff_ffff
       result = initRand(time)
     else:
-      let now = times.getTime()
-      result = initRand(convert(Seconds, Nanoseconds, now.toUnix) + now.nanosecond)
+      proc getRandomState(): Rand =
+        when defined(nimscript):
+          result = Rand(
+            a0: CompileTime.hash.Ui,
+            a1: CompileDate.hash.Ui)
+          if not result.isValid:
+            result = DefaultRandSeed
+        else:
+          var urand: array[sizeof(Rand), byte]
+
+          for i in 0 .. 7:
+            if sysrand.urandom(urand):
+              copyMem(result.addr, urand[0].addr, sizeof(Rand))
+              if result.isValid:
+                break
+
+          if not result.isValid:
+            # When 2 processes executed at same time on different machines,
+            # `result` can still have same value.
+            let
+              pid = getCurrentProcessId()
+              t = getMonoTime().ticks
+            result.a0 = pid.hash().Ui
+            result.a1 = t.hash().Ui
+            if not result.isValid:
+              result.a0 = pid.Ui
+              result.a1 = t.Ui
```

Comment:
This code was probably never tested and is a liability for security.

Reply:
So `baseState` should be initialized only with `sysrand.urandom` and if it failed, just quit or raise error?

Reply:
Yes, just quit.

Reply:
Done!